templates/repo/issue/view_content.tmpl

```diff
@@ -85,7 +85,7 @@
 					</div>
 					{{$reactions := .Issue.Reactions.GroupByType}}
 					{{if $reactions}}
-						<div class="ui attached segment reactions">
+						<div class="ui attached segment reactions" role="complementary">
```

Comment:
`complementary` for reactions?
This doesn't seem right, if I read the MDN docs correctly…

Reply:
probably should be `comment` AFAICS

Reply:
I agree that [is not good for this](https://developer.mozilla.org/en-US/docs/Web/Accessibility/ARIA/Roles/complementary_role#accessibility_concerns). Landmark role [it is not as good practice to use non-semantic elements, such as `<div>`, adding semantics with landmark roles](https://developer.mozilla.org/en-US/docs/Web/Accessibility/ARIA/Roles/landmark_role#best_practices).
Instead, I would suggest a [menu role](https://developer.mozilla.org/en-US/docs/Web/Accessibility/ARIA/Roles/menu_role).

Reply:
> probably should be `comment` AFAICS

comment role should be for the whole comment container (line 29)

Reply:
@fsologureng Gitea is already overusing the `menu` role, it's full of unlabeled menus. Actually what I would do is to combine the reactions with the comment itself, so it would be in the same `article`, but I'm not confident enough that if I wrap the whole thing in one more `div`, it won't upset the visuals.  
First I worried that I didn't know about the `comment` role, but then I saw why. It's actually in a _draft_ state for ARIA 1.3, meaning that probably no screen reader supports it yet. And yes, even when it gets out of draft, it would be nice to mark the whole comment up with it.  
I agree that `complementary` is not the best fit for reactions. What about `note` then?

Reply:
> A `note` role suggests a section whose content is parenthetic or ancillary to the main content.

Yep, sounds fine.

Reply:
> @fsologureng Gitea is already overusing the `menu` role, it's full of unlabeled menus. Actually what I would do is to combine the reactions with the comment itself, so it would be in the same `article`, but I'm not confident enough that if I wrap the whole thing in one more `div`, it won't upset the visuals. First I worried that I didn't know about the `comment` role, but then I saw why. It's actually in a _draft_ state for ARIA 1.3, meaning that probably no screen reader supports it yet. And yes, even when it gets out of draft, it would be nice to mark the whole comment up with it. I agree that `complementary` is not the best fit for reactions. What about `note` then?

Yes @Menelion, that bad design of menus should change. I have a PR repairing the label issue of reactions, but is not ready yet. I think both changes could be complementary.

Understood your apprehension about `comment`, I didn't aware of that.

`note` is for content and for complement main resource content. Reactions, instead, are essentially a set of actionable things. What about [`group`](https://www.w3.org/TR/wai-aria-1.1/#group), which is more generic but allows a differentiation with the rest of the comment body. I think it should works.